docs/developers/debugging.md

```diff
@@ -0,0 +1,133 @@
+
+Running Fuel Core with CLion requires the official JetBrains Rust plugin. Find the Rust plugin by navigating to CLion's Plugin menu, entering "Rust" in the search. Install it by clicking "Install".
+
+### Create A Run Configuration
+
+1. In the top-right corner of the CLion window, click on the configurations dropdown menu and select "Edit Configurations..."
+2. In the "Run/Debug Configurations" window, click on the "+" button and select "Cargo"
+3. In the "Name" field, give your configuration a descriptive name, such as "Run Beta 3"
+4. In the "Command" field, enter `run --all-features --bin fuel-core -- run <ARGUMENTS>`, where `<ARGUMENTS>` represents the desired CLI arguments. Refer to the CLI documentation for guidance on which options and arguments to use.
+5. If you use environment variables for configuration (as indicated in the CLI documentation), you can set them in the "Environment variables" section. Enter the variables and their values as needed. It is recommended to set the `RUST_LOG` environment variable to either `info` or `debug`.
+6. Click "Apply" or "OK" to save your run configuration.
+
+### Example Configuration
+
+![clion_configuration.png](assets/clion_configuration.png)
+
+### Running and Debugging
+
+1. With your configuration selected in the dropdown menu, click on the Run button to start running your Fuel client node.
+2. Alternatively, click on the Debug button to start your client node in debug mode.
+3. Set breakpoints in the code where inspections are necessary. CLion will halt execution at these breakpoints, allowing variable inspection and step-by-step code execution.
+
+## Visual Studio Code
+
+### Prerequisites
+
+Running Fuel Core with Visual Studio Code requires the CodeLLDB extension. Find the CodeLLDB extension by navigating to the "Extensions" tab and searching "CodeLLDB". Install it by clicking "Install".
+
+### Create A Run Configuration
+
+1. In VS Code, click on the "Run and Debug" icon in the left sidebar.
+2. VS Code will offer to create a `launch.json` file to configure how your client node is run. Clicking this will prompt you to select a debugger. Select the CodeLLDB debugger to proceed. VS Code will generate a `launch.json` file for you.
+3. In the "Run and Debug" menu, click the dropdown menu and select "Add Configuration..."
+4. In the `"type"` field, enter `"lldb"`. In the `"request"` field, enter `"launch"`.
+5. In the `"name"` field, give your configuration a descriptive name, such as `"Run Beta 3"`
+6. In the `"args"` array nested under `"cargo"`, enter the following arguments: `"run"`, `"--all-features"`, `"--bin"`, `"fuel-core"`, `"--"`, `"run"`, `<ARGUMENTS>`, where `<ARGUMENTS>` is a list of CLI values and arguments encapsulated by quotes. Refer to the CLI documentation for guidance on which options and arguments to use.
+7. Save these changes made to your `launch.json` file.
+
+### Example Configuration
+
+![vs_code_configuration.png](assets/vs_code_configuration.png)
```

Comment:
We could just past here the plain json, to make it easier to copy and paste

Reply:
I was not able to make the debug work, it looks like the command never finishes. It starts but not finishes.

My config;
```
[
    "run",
    "--all-features",
    "--bin",
    "fuel-core",
    "--",
    "run",
    "--ip", "0.0.0.0",
    "--port", "4004",
    "--db-type", "in-memory",
    "--utxo-validation",
    "--vm-backtrace",
    "--poa-interval-period",
    "30s"
]
```

Reply:
Luiz and I discussed this specific issue over a call: It appears that the client is starting correctly, and producing the expected output in the terminal, but the VS Code debugger is not correctly attaching to the running process. Luiz and I will spend some time independently looking into this.

Reply:
I updated the example with a more reliable config that users can copy + paste to get started.